tensorflow/c/kernels_experimental.cc

```diff
@@ -331,6 +332,11 @@ void TF_GetInputTensorFromVariable(TF_OpKernelContext* ctx, int input,
                                                     TF_Tensor* dest),
                                    TF_Tensor** out, TF_Status* status) {
   auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(ctx);
+
+  auto status_setter = ::tensorflow::gtl::MakeCleanup([cc_ctx]() {
```

Comment:
It is. It's a meant to automatically set the `status` variable when we return from the function at any point (e.g. when we call `OP_REQUIRES_OK`).

Reply:
I see.

As a side note, the error reporting in those `TF_*` APIs is so messy. First there is the duplication between reporting via `ctx` and reporting via `status`. Then even if we choose to keep `status` it should be a return value instead of an output argument. (CCing @penpornk)

Reply:
I agree, it's pretty messy and we keep expanding that messiness to make all API endpoints fit together. I wouldn't mind getting rid of the error reporting via `ctx` and just returning an error status instead.